### DISCUSSION TITLE
Can't resolve click dependency versions?

### DISCUSSION THREAD

**Discussion by awinder** [apache/airflow#24760] [Q&A]
Hello,

Apologies if I'm missing something basic here, but we're installing airflow & locking dependencies using a requirements.in file that has 2 airflow dependencies specified as:

apache-airflow[crypto,celery,postgres,hive,jdbc,mysql,ssh]==2.3.2
apache-airflow-providers-http==3.0.0

When pip tries to resolve dependencies of airflow though it seems to get caught up inside of airflow subpackages, specifically on the click package:

```
Could not find a version that matches click<8.0,<9.0,>=5.1,>=6.7,>=8.0.3 (from flask==1.1.4->apache-airflow[celery,crypto,hive,jdbc,mysql,postgres,ssh]==2.3.2->-r requirements.in (line 6))
Tried: 0.1, 0.1, 0.2, 0.2, 0.3, 0.3, 0.4, 0.4, 0.5, 0.5, 0.5.1, 0.5.1, 0.6, 0.6, 0.7, 0.7, 1.0, 1.0, 1.1, 1.1, 2.0, 2.0, 2.1, 2.1, 2.2, 2.2, 2.3, 2.3, 2.4, 2.4, 2.5, 2.5, 2.6, 2.6, 3.0, 3.0, 3.1, 3.1, 3.2, 3.2, 3.3, 3.3, 4.0, 4.0, 4.1, 4.1, 5.0, 5.0, 5.1, 5.1, 6.0, 6.0, 6.1, 6.1, 6.2, 6.2, 6.3, 6.3, 6.4, 6.4, 6.5, 6.6, 6.6, 6.7, 6.7, 7.0, 7.0, 7.1, 7.1, 7.1.1, 7.1.1, 7.1.2, 7.1.2, 8.0.0, 8.0.0, 8.0.1, 8.0.1, 8.0.2, 8.0.2, 8.0.3, 8.0.3, 8.0.4, 8.0.4, 8.1.0, 8.1.0, 8.1.1, 8.1.1, 8.1.2, 8.1.2, 8.1.3, 8.1.3
Skipped pre-versions: 6.7.dev0, 8.0.0a1, 8.0.0a1, 8.0.0rc1, 8.0.0rc1
There are incompatible versions in the resolved dependencies:
  click<9.0,>=8.0.3 (from celery==5.2.7->apache-airflow[celery,crypto,hive,jdbc,mysql,postgres,ssh]==2.3.2->-r requirements.in (line 6))
  click<8.0,>=5.1 (from flask==1.1.4->apache-airflow[celery,crypto,hive,jdbc,mysql,postgres,ssh]==2.3.2->-r requirements.in (line 6))
  click<9,>=6.7 (from flask-appbuilder==3.4.5->apache-airflow[celery,crypto,hive,jdbc,mysql,postgres,ssh]==2.3.2->-r requirements.in (line 6))
 ```

Pulling the celery package out of apache-airflow seems to resolve the issue, because the version that conflicts with most of the other airflow subpackages is defined from the celery dependency. Anyways -- is there some recommended way to overcome this? Or does this need version wrangling in the airflow packages themselves?

**Comment by potiuk** [comment]
You should use constraints when installing airflow: https://airflow.apache.org/docs/apache-airflow/stable/installation/installing-from-pypi.html
